src/net.cpp

```diff
@@ -118,6 +118,32 @@ void CConnman::AddAddrFetch(const std::string& strDest)
 
 uint16_t GetListenPort()
 {
+    // If -bind= is provided with ":port" part, use that (first one if multiple are provided).
+    for (const std::string& bind_arg : gArgs.GetArgs("-bind")) {
+        constexpr uint16_t dummy_port = 0;
+        constexpr bool allow_lookup = false;
+        CService bind_addr;
+
+        if (Lookup(bind_arg, bind_addr, dummy_port, allow_lookup)) {
+            if (bind_addr.GetPort() != dummy_port) {
+                return bind_addr.GetPort();
+            }
+        }
+    }
+
+    // Otherwise, if -whitebind= without "noban" is provided, use that
```

Comment:
nit, while updating here could be consistent with 7075f604e8d0b21
 ```suggestion
    // Otherwise, if -whitebind= without NetPermissionFlags::NoBan permission is provided, use that
```

Reply:
Done.